CDCgov/prime-reportstream
# Bug: fix error mapping NTE segments in ORM messages

This PR fixes #13626. 

Test Steps:
1. Send an `ORM` message that has `NTE` segments to the waters endpoint using the `flexion.simulated-hospital` client. You can try using this ORM:
    ```
   MSH|^~\&|BaptistOracle^2.16.840.1.114222.4.1.000000^ISO|BaptistEast^2.16.840.1.114222.4.1.000001^ISO|ALlabNatus^2.16.840.1.114222.4.1.181960.2^ISO|ALlab^simulated-lab-id^ISO|20240224134009||ORM^O01^ORM_O01|Q1960841872T2476960690||2.5.1||||||8859/1
    PID|1||1300974^^^Baptist East^MR||ONE^TESTCASE||202402210152-0500|F^Female^HL70001||2106-3^White^HL70005|1234 GPCS WAY^^MONTGOMERY^Alabama^36117^USA^home^^Montgomery|||||||2227600015||||N^Not Hispanic or Latino^HL70189|||1|||||
    NK1|1|ONE^MOMFIRST|MTH^Mother^HL70063||^^^^^804^5693861||||||||||||||||||||||||||||123456789^^^Medicaid&2.16.840.1.113883.4.446&ISO^MD||||000-00-0000^^^ssn&2.16.840.1.113883.4.1&ISO^SS
    ORC|NW|2801690163^ORDERID||||||||||12345^^^^^^^^NPI&2.16.840.1.113883.4.6&ISO^L|||||||||
    OBR|1|2801690163^ORDERID||54089-8^Newborn screening panel AHIC^LN|||202402221854-0500|||||||||12345^^^^^^^^NPI&2.16.840.1.113883.4.6&ISO^L||||||||
    OBX|1|ST|57723-9^Unique bar code number of Current sample^LN||123456||||||F|||202402221854-0500
    NTE|1||note 1|
    NTE|2||note 2|
    ```
    `curl --header 'Content-Type: application/hl7-v2' --header 'Client: flexion.simulated-hospital' --header 'Authorization: Bearer token' --data-binary '@/path/to/etor_ORM_message.hl7' 'http://localhost:7071/api/waters'`
3. Verify the message was delivered to its final destination without throwing an error when converting FHIR to HL7

## Checklist

### Testing
- [ ] Ran `./gradlew testSmoke`

## Linked Issues
- Fixes #13626
